### ISSUE TITLE
[attr-defined] only checked for `self` when `self` is explicitly annotated

### ISSUE THREAD

**Issue by Tetratrio** [python/mypy#9521]
I can not get mypy to give me an error for the following code:
```python
class A:
    value: str = 'hello'

    def f(self):
        print(self.value)

    def x(self):
        print(self.other_value)
```
I was hoping that mypy would be able to find the error.
```
Success: no issues found in 1 source file
```
If I explicitly annotate the `self` argument to function `A.x()` to `'A'` mypy will actually report the error.
```python
class A:
    value: str = 'hello'

    def f(self):
        print(self.value)

    def x(self: 'A'):
        print(self.other_value)
```
mypy gives
```
test.py:44: error: "A" has no attribute "other_value"
Found 1 error in 1 file (checked 1 source file)
```

I am a bit confused about how this can be the case when in [PEP 484](https://www.python.org/dev/peps/pep-0484/#the-meaning-of-annotations) it is stated that the first argument of instance method implicitly have the type of the class they are defined under.

**Comment by gvanrossum** [comment]
Mypy skips functions that don't have any annotations.

For more help please see Gitter.
